pkg/util/winutil/process.go

```diff
@@ -0,0 +1,276 @@
+// +build windows
+
+package winutil
+
+import (
+	"fmt"
+	"unsafe"
+
+	"golang.org/x/sys/windows"
+)
+
+var (
+	modntdll                      = windows.NewLazyDLL("ntdll.dll")
+	modkernel                     = windows.NewLazyDLL("kernel32.dll")
+	procNtQueryInformationProcess = modntdll.NewProc("NtQueryInformationProcess")
+	procReadProcessMemory         = modkernel.NewProc("ReadProcessMemory")
+	procIsWow64Process            = modkernel.NewProc("IsWow64Process")
+)
+
+// C definition from winternl.h
+
+//typedef enum _PROCESSINFOCLASS {
+//    ProcessBasicInformation = 0,
+//    ProcessDebugPort = 7,
+//    ProcessWow64Information = 26,
+//    ProcessImageFileName = 27,
+//    ProcessBreakOnTermination = 29
+//} PROCESSINFOCLASS;
+
+type PROCESSINFOCLASS uint32
+
+const (
+	ProcessBasicInformation   = PROCESSINFOCLASS(0)
+	ProcessDebugPort          = PROCESSINFOCLASS(7)
+	ProcessWow64Information   = PROCESSINFOCLASS(26)
+	ProcessImageFileName      = PROCESSINFOCLASS(27)
+	ProcessBreakOnTermination = PROCESSINFOCLASS(29)
+)
+
+// IsWow64Process determines if the specified process is running under WOW64
+// that is, if it's a 32 bit process running on 64 bit winodws
+func IsWow64Process(h windows.Handle) (is32bit bool, err error) {
+	is32bit = false
+	err = nil
+	var wow64Process uint32
+
+	r, _, _ := procIsWow64Process.Call(uintptr(h),
+		uintptr(unsafe.Pointer(&wow64Process)))
+
+	if r == 0 {
+		return false, windows.GetLastError()
+	}
+	if wow64Process == 0 {
+		is32bit = false
+	} else {
+		is32bit = true
```

Comment:
nit: If I understand correctly, it's only when `wow64Process > 0` then `is32bit` is `true` right?

Reply:
wow64Process is a C `BOOL` type.  So zero is false, and any non-zero is true

Reply:
I mean in golang bool default value is false, you could just say `if wow64Process > 0 {is32bit = true}`. Not a big deal.